devenv/aws-setup/initialize.py

```diff
@@ -0,0 +1,72 @@
+import boto3
+import os
+import json
+from collections import defaultdict
+
+# Setup Script
+# ------------
+
+# Extract environment variables.
+input_template_file = os.environ["INPUT_CDK_TEMPLATE"]
+output_template_file = os.environ["OUTPUT_CDK_TEMPLATE"]
+dynamodb_endpoint = os.environ["DYNAMODB_ENDPOINT"]
+local_lambda_path = os.environ["LOCAL_LAMBDA_PATH"]
+
+# Setup DynamoDB Client.
+client = boto3.client('dynamodb', endpoint_url=dynamodb_endpoint)
+
+# Read CDK Template.
+with open(input_template_file) as fp:
+    template = json.load(fp)
+
+# Create convenience mapping.
+template_resource_ids_for_type = defaultdict(list)
```

Comment:
The resource ids used as a key for a resource in the template is actually a unique token that is associated with the resource for within the template, but there's a separate AWS resource id, which is why this refers to `template_resource_id`.

`template_resource_token` could be a better name, but it is an identifier.